Service/src/main/java/Repository/UserRepositoryImplPostgres.java

```diff
@@ -0,0 +1,376 @@
+                }
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return trainers;
+    }
+
+    @Override
+    public HashMap<Integer, UserImpl> allStudent() {
+        HashMap<Integer, UserImpl> students = new HashMap<>();
+        try (Connection connection = datasourse.getConnection()){
+            PreparedStatement ps =  connection.prepareStatement("select * from users where role_id=3;");
+            ResultSet rs = ps.executeQuery();
+            while (rs.next()) {
+                if (!rs.wasNull() || !rs.getString("login").equals("")) {
+                Student student = new Student();
+                Integer studentId = rs.getInt("id");
+                    if (!rs.rowDeleted())
+                students.put(studentId,
+                        student
+                                .withTheamMark(new HashMap<>())
+                                .withId(studentId)
+                                .withLogin(rs.getString("login"))
+                                .withPassword(rs.getString("password"))
+                                .withName(rs.getString("name"))
+                                .withAge(rs.getInt("age"))
+                                .withRole(checkRole(rs.getInt("role_id")))
+                );
+                }
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return students;
+    }
+
+    @Override
+    public HashMap<Integer, UserImpl> allAdmin() {
+        HashMap<Integer, UserImpl> administrators = new HashMap<>();
+        try (Connection connection = datasourse.getConnection()){
+            PreparedStatement ps =  connection.prepareStatement("select * from users where role_id=1;");
+            ResultSet rs = ps.executeQuery();
+            while (rs.next()) {
+                if (!rs.wasNull() || !rs.getString("login").equals("")) {
+                Administrator administrator = new Administrator();
+                Integer adminId = rs.getInt("id");
+                    if (!rs.rowDeleted())
+                administrators.put(adminId,
+                        administrator
+                                .withId(adminId)
+                                .withLogin(rs.getString("login"))
+                                .withPassword(rs.getString("password"))
+                                .withName(rs.getString("name"))
+                                .withAge(rs.getInt("age"))
+                                .withRole(checkRole(rs.getInt("role_id")))
+
+                );}
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return administrators;
+    }
+
+    @Override
+    public UserImpl getUserById(Integer id) {
+        UserImpl user =  new UserImpl();
+        try (Connection connection = datasourse.getConnection()){
+            PreparedStatement ps = connection.prepareStatement("SELECT * FROM users where id="+"(?)");
```

Comment:
Странная строка запроса. А почему не сделать сразу вот так:
```suggestion
            PreparedStatement ps = connection.prepareStatement("SELECT * FROM users where id=?");
```

Reply:
Исправил.